### PR TITLE
Remove docker override

### LINKED ISSUE #252: `docker-compose.override.yml` file breaks the `compose` sub cmd
The override file must match the compose file format version of the main `docker-compose.yml` file.
While the main `docker-compose.yml` file is managed by `sourced compose`, the contents of the `docker-compose.override.yml` file are hardcoded into the `sourced` bin.

See for example the upgrading instructions here https://github.com/src-d/sourced-ce/pull/251/files

Why is the `docker-compose.override.yml` file needed at all in `sourced`? Can we find a way to manage this cleanly from the preparation scripts of `sourced-ui` instead?

### PR DESCRIPTION
**Pull request by se7entyse7en**
Closes #252.

---

* [x] I have updated the CHANGELOG file according to the conventions in [keepachangelog.com](https://keepachangelog.com)
* [ ] This PR contains changes that do not require a mention in the CHANGELOG file

### CODE REVIEW

**Comment by smacker** [comment]
something wrong with go modules. This PR is introducing new dependency but shouldn't as it only deletes code.

**Comment by se7entyse7en** [comment]
> something wrong with go modules. This PR is introducing new dependency but shouldn't as it only deletes code.

yup sorry, I opened this just before leaving and there was some previous experiment leftover 😅.

**Review by smacker — APPROVED** [review]

**Review by dpordomingo — APPROVED** [review]
thanks!

File: cmd/sourced/compose/workdir/handler.go [hunk 1]
```diff
@@ -152,7 +152,7 @@ func (h *Handler) Remove(w *Workdir) error {
 
 	basePath := filepath.Join(h.workdirsPath, subPath)
 
-	for _, f := range append(RequiredFiles, OptionalFiles...) {
+	for _, f := range RequiredFiles {
```
Comment (dpordomingo) [inline cmd/sourced/compose/workdir/handler.go:155]: I like this

File: cmd/sourced/compose/workdir/workdir.go [hunk 2]
```diff
@@ -20,9 +20,6 @@ var (
 	// RequiredFiles list of required files in a directory to treat it as a working directory
 	RequiredFiles = []string{".env", "docker-compose.yml"}
 
-	// OptionalFiles list of optional files that could be deleted when pruning
-	OptionalFiles = []string{"docker-compose.override.yml"}
```
Comment (dpordomingo) [inline cmd/sourced/compose/workdir/workdir.go:24]: And this

**Review by dpordomingo — COMMENTED** [review]
Review on closed PR

**Pull request merged**